Library/ViewModels/CommentsViewModel.swift

```diff
@@ -22,15 +34,52 @@ public final class CommentsViewModel: CommentsViewModelType,
   CommentsViewModelInputs,
   CommentsViewModelOutputs {
   public init() {
-    // FIXME: Configure this VM with a project in order to feed the slug in here to fetch comments
-    // Call this again with a cursor to paginate.
-    self.viewDidLoadProperty.signal.switchMap { _ in
-      AppEnvironment.current.apiService
-        .fetchComments(query: comments(withProjectSlug: "bring-back-weekly-world-news"))
-        .ksr_delay(AppEnvironment.current.apiDelayInterval, on: AppEnvironment.current.scheduler)
-        .materialize()
+    let projectOrUpdate = Signal.combineLatest(
+      self.projectAndUpdateProperty.signal.skipNil(),
+      self.viewDidLoadProperty.signal
+    )
+    .map(first)
+    .flatMap { project, update in
+      SignalProducer(value: project.map(Either.left) ?? update.map(Either.right))
+        .skipNil()
     }
-    .observeValues { print($0) }
+
+    let initialProject = projectOrUpdate
+      .flatMap { projectOrUpdate in
+        projectOrUpdate.ifLeft(
+          SignalProducer.init(value:),
+          ifRight: {
+            AppEnvironment.current.apiService.fetchProject(param: .id($0.projectId)).demoteErrors()
+          }
+        )
+      }
+
+    let isCloseToBottom = self.willDisplayRowProperty.signal.skipNil()
+      .map { row, total in row >= total - 3 }
```

Comment:
I wonder what number we should use here for the smoothest experience. It felt a little too close to the bottom when I was doing some testing on the simulator. What do you think?

cc: @justinswart

Reply:
Yea I just used what `DeprecatedCommentsViewModel` was previously using, so definitely can be higher given the actual card itself will be much taller than the cell being used now.

Reply:
Yeah value feels arbitrary,  we should use whatever feels right.